### PR TITLE
Colin git api

### PR DESCRIPTION
**Pull request by coloniusrex**
-Creates Repo PORO for Github User display section
- Creates Github User Dashboard Display, will not display if user does not have token
-Adds token to Users table 
-Creates Faraday connection for GitHub API's
-Connects to repos and user resources
-Refactors conditional statement in UserVideos 

-All tests passing

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,17 @@
 class UsersController < ApplicationController
-  def show; end
+  def show
+    conn = Faraday.new(url: "https://api.github.com") do |faraday|
+      faraday.headers['authorization'] = "token #{ENV["GITHUB_TOKEN"]}"
+    end
+    response = conn.get('/user/repos')
+    user_response = conn.get('/user')
+    json = JSON.parse(response.body, symbolize_names: true)
+    user_json = JSON.parse(user_response.body, symbolize_names: true)
```
Comment (coloniusrex) [inline app/controllers/users_controller.rb:9]: Curious to see what we can find for refactoring, reducing requests. WIll find out soon!

File: app/controllers/user_videos_controller.rb [hunk 2]
```diff
@@ -2,15 +2,13 @@ class UserVideosController < ApplicationController
...
+    user_video = UserVideo.new(user_video_params)
+    if current_user == nil
       flash[:error] = 'You need to login to bookmark videos!'
+    elsif current_user.user_videos.find_by(video_id: user_video.video_id)
+      flash[:error] = 'Already in your bookmarks'
+    elsif user_video.save
+      flash[:success] = 'Bookmark added to your dashboard!'
     end
     redirect_back(fallback_location: root_path)
   end
```
Comment (SMJ289) [inline app/controllers/user_videos_controller.rb:14]: This logic looks cleaner than before! I don't know why the guard clause didn't work...

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+    @github_username = user_json[:login]
+    @repos = json.map do |repo_data|
+      Repo.new(repo_data)
+    end
+  end
```
Comment (SMJ289) [inline app/controllers/users_controller.rb:15]: We will need to refactor this into some helper methods at some point. We should add an issue on the project board.

Comment (SMJ289) [reply app/controllers/users_controller.rb:15]: Maybe we can reduce the logic to use just one instance variable?

**Pull request merged**